pkg/server/server.go

```diff
@@ -211,6 +211,14 @@ func coreControllers(ctx context.Context, sc *Context, config *Config) error {
 		helmchart.DefaultJobImage = config.ControlConfig.SystemDefaultRegistry + "/" + helmchart.DefaultJobImage
 	}
 
+	if config.ControlConfig.KlipperHelmImage != "" {
+		helmchart.DefaultJobImage = config.ControlConfig.KlipperHelmImage
+	}
+
+	if config.ControlConfig.SystemDefaultRegistry != "" && config.ControlConfig.KlipperHelmImage != "" {
+		helmchart.DefaultJobImage = config.ControlConfig.SystemDefaultRegistry + "/" + config.ControlConfig.KlipperHelmImage
+	}
```

Comment:
Only apply the system default registry if a custom job image hasn't been set; the custom job image should take precedence.
```suggestion
	if config.ControlConfig.HelmJobImage != "" {
		helmchart.DefaultJobImage = config.ControlConfig.HelmJobImage
	} else if config.ControlConfig.SystemDefaultRegistry != "" {
		helmchart.DefaultJobImage = config.ControlConfig.SystemDefaultRegistry + "/" + helmchart.DefaultJobImage
	}
```